lib/models/order.rb

```diff
@@ -16,8 +16,11 @@ class Order
 
   def self.placed_in(office)
     orders = all(:date => Days.from_monday_to_friday)
-    crafters = User.all(:office => office)
-    orders.keep_if { |o| crafters.any? { |c| c.slack_id == o.user_id || c.slack_id == o.host } }
+    users = User.all(:office => office)
+
+    orders.keep_if do |order|
+      users.any? { |user| user.slack_id == order.user_id || user.slack_id == order.host }
```

Comment:
Nice one, I wonder if we could push it to: 
``` ruby
orders.keep_if { |order| users.any? { |user| has_ordered(order, user) } }
```
That would split the responsibility of filtering and the logic, I'm not quite sure if it make it easier to follow

Reply:
Not sure. My gut feeling here is that we are fighting against how the relationships are modelled. This might need more thought.